src/client/Microsoft.Identity.Client/ApiConfig/AbstractPublicClientAcquireTokenParameterBuilder.cs

```diff
@@ -35,6 +36,37 @@ public override Task<AuthenticationResult> ExecuteAsync(CancellationToken cancel
             return ExecuteInternalAsync(cancellationToken);
         }
 
+        /// <summary>
+        ///  Modifies the token acquisition request so that the acquired token is a Proof of Possession token (PoP), rather than a Bearer token. 
+        ///  PoP tokens are similar to Bearer tokens, but are bound to the HTTP request and to a cryptographic key, which MSAL can manage on Windows.
+        ///  Note that only the host and path parts of the request URI will be bound.
+        ///  See https://aka.ms/msal-net-pop
+        /// </summary>
+        /// <param name="httpMethod">The HTTP method ("GET", "POST" etc.) method that will be bound to the token. Leave null and the POP token will not be bound to the method.
+        /// Corresponds to the "m" part of the a signed HTTP request.</param>
+        /// <param name="requestUri">The URI to bind the signed HTTP request to.</param>
+        /// <param name="nonce">Nonce of the protected resource (RP) which will be published as part of the WWWAuthenticate header associated with a 401 HTTP response
+        /// or as part of the AuthorityInfo header associated with 200 response. Set it here to make it part of the Signed HTTP Request part of the POP token.</param>
+        /// <returns>The builder.</returns>
+        /// <remarks>
+        /// <list type="bullet">
+        /// <item><description>An Authentication header is automatically added to the request</description></item>
+        /// <item><description>The PoP token is bound to the HTTP request, more specifically to the HTTP method (GET, POST, etc.) and to the Uri (path and query, but not query parameters).</description></item>
+        /// <item><description>MSAL creates, reads and stores a key in memory that will be cycled every 8 hours.</description></item>
+        /// <item><description>This is an experimental API. The method signature may change in the future without involving a major version upgrade.</description></item>
+        /// </list>
+        /// </remarks>
+        public T WithProofOfPossession(HttpMethod httpMethod, Uri requestUri, string nonce)
+        {
```

Comment:
Wouldn't a resource throw exception anyway since it would expect a POP token, but would get a Bearer?

Reply:
No, I don't think any resource out there will support just POP. They will support Bearer and POP. However, a CA policy may prevent a Bearer token from gaining access.

Reply:
I think this decision point warrants a bigger discussion with MSAL C++ team, the aim is to be consistent with them. 

CC @rayluo and @siddhijain and @sameerag - the context here is "POP is only supported if WAM is present on the box, i.e. Win10. If on Win8, Mac or Linux and app developer configures POP, what should MSAL do? Return a bearer token or throw?"

Reply:
I think the library should throw an exception to let the user know that their device does not support PoP. If we return a bearer, they will never know that they have not received the right kind of token.

Reply:
@siddhijain - yes, erroring out soon is a good point, but I am not sure if it will help with the dev experience. The dev experience would be something like: 

```csharp
var pca = new PubliClientApplication().PreferBroker();

if (pca.IsBrokerAvailable)
{
     pca.AcquireTokenInteractive().WithProofOfPosession("http://www.bing.com", nonce);
}
else
{
    pca.AcquireTokenInteractive();
}

```

Reply:
I think the concern is if someone wants to use a more secure PoP token but gets less secure Bearer without knowing. We'd have to suggest for them to monitor the token type. This can be forced by the resource to only accept PoP tokens. If the resource accepts both token types, then having this method fallback to Bearer might not be as bad. Also, we could later add an overload with a boolean parameter whether to fallback or not.